lib/interceptor/proxy.js

```diff
@@ -0,0 +1,11 @@
+'use strict'
+const ProxyAgent = require('../dispatcher/proxy-agent')
+
+module.exports = opts => {
+  const agent = new ProxyAgent(opts)
+  return () => {
+    return function proxyInterceptor (opts, handler) {
+      return agent.dispatch(opts, handler)
+    }
+  }
+}
```

Comment:
This isn't an interceptor?

Reply:
That's what I was referring to when I suggested that `ProxyAgent` works differently 😅 
It requires to hope over the whole dispatch, as it forwards the request to the proxy instead of the intended dispatch.
It does not respects the previous dispatcher, so if proxy is used, it should be placed at the very beginning of the chain.

I couldn't think of any other way without re-working the `ProxyAgent`, but totally open to suggestions

Reply:
I will have to investigate this

Reply:
It sounds to me like this should not be a interceptor.

Reply:
Let's skip this for now so we can land.